File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.go
func findMin(nums []int) int {
	if len(nums) ==1 {
		return nums[0]
	}
	left:=0
	right:=len(nums)-1
	for left <=right {
		next:=left+1
		if next < len(nums) && nums[left] > nums[next] {
			return nums[next]
		}
		left++
	}
	return nums[0]
	

}

File: Data Structures & Algorithms/sort-an-array/submission-0.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        

        def merge_sort(nums):
            if len(nums) <= 1:
                return nums
            mid=len(nums)//2
            left=merge_sort(nums[:mid])
            right=merge_sort(nums[mid:])
            return merge(left,right)
        def merge(left,right):
            result=[]
            i,j=0,0
            while i < len(left) and j < len(right):
                if left[i] > right[j]:
                    result.append(right[j])
                    j+=1
                else:
                    result.append(left[i])
                    i+=1
            result.extend(left[i:]) # append remaining elements
            result.extend(right[j:])
            return result
        return merge_sort(nums)